InterpolatorTest fails on windows when PATH is the first environment variable

When running **tests** with **maven** on **windows** _InterpolatorTest_ fails when **PATH** is the first environment variable

Not sure if its a problem with the `Interpolator` or just the test setup.

```[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.weareadaptive.nexus.casc.plugin.internal.InterpolatorTest
[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.283 s <<< FAILURE! - in com.weareadaptive.nexus.casc.plugin.internal.InterpolatorTest
[ERROR] interpolateWithEnvVar  Time elapsed: 0.247 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <hello C:\Program Files\Amazon Corretto\jdk11.0.11_9\bin;C:\Program Files\[...]
	but was: <hello C:Program FilesAmazon Correttojdk11.0.11_9bin;C:Program Files[...]
	at com.weareadaptive.nexus.casc.plugin.internal.InterpolatorTest.interpolateWithEnvVar(InterpolatorTest.java:22)
```

--- a/pom.xml
+++ b/pom.xml
@@ -24,6 +24,7 @@
         <snakeyaml.version>1.26</snakeyaml.version>
 
         <junit-jupiter.version>5.6.2</junit-jupiter.version>
+        <system-lambda.version>1.2.0</system-lambda.version>
     </properties>
 
     <dependencies>
@@ -76,6 +77,12 @@
             <version>${snakeyaml.version}</version>
         </dependency>
 
+        <dependency>
+            <groupId>com.github.stefanbirkner</groupId>
+            <artifactId>system-lambda</artifactId>
+            <version>${system-lambda.version}</version>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-engine</artifactId>
@@ -233,8 +240,8 @@
                     </execution>
                 </executions>
             </plugin>
-        </plugins>
-    </build>
+                </plugins>
+            </build>
 
     <repositories>
         <repository>

--- a/src/test/java/com/weareadaptive/nexus/casc/plugin/internal/InterpolatorTest.java
+++ b/src/test/java/com/weareadaptive/nexus/casc/plugin/internal/InterpolatorTest.java
@@ -2,28 +2,41 @@ package com.weareadaptive.nexus.casc.plugin.internal;
 
 import org.junit.jupiter.api.Test;
 
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.Callable;
 
+import static com.github.stefanbirkner.systemlambda.SystemLambda.withEnvironmentVariable;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 class InterpolatorTest {
 
     @Test
     void interpolateWithFile() {
-        assertEquals("hello world", new Interpolator().interpolate("hello ${file:"+getClass().getClassLoader().getResource("test").getPath()+"}"));
+        assertEquals("hello world", new Interpolator().interpolate("hello ${file:" + getClass().getClassLoader().getResource("test").getPath() + "}"));
     }
 
     @Test
-    void interpolateWithEnvVar() {
-        Map.Entry<String, String> envVar = System.getenv().entrySet().iterator().next();
-        String key = envVar.getKey();
-        String value = envVar.getValue();
+    void interpolateWithEnvVar() throws Exception {
+        final String NEXUS_TEST_ENV_VAR = "NEXUS_TEST_ENV_VAR";
+        withEnvironmentVariable(NEXUS_TEST_ENV_VAR, "any Value")
+                .execute((Callable<List<String>>) () -> {
+                    final Map.Entry<String, String> entry = System.getenv()
+                            .entrySet()
+                            .stream()
+                            .filter(e -> NEXUS_TEST_ENV_VAR.equals(e.getKey()))
+                            .findFirst()
+                            .orElseThrow(() -> new RuntimeException("Mocking environment variables failed."));
 
-        assertEquals("hello " + value, new Interpolator().interpolate("hello $"+key));
-        assertEquals("hello " + value, new Interpolator().interpolate("hello ${"+key+"}"));
-        assertEquals("hello " + value, new Interpolator().interpolate("hello ${"+key+":\"\"}"));
-        assertEquals("hello " + value, new Interpolator().interpolate("hello ${"+key+":}"));
-        assertEquals("hello " + value, new Interpolator().interpolate("hello ${"+key+":foo}"));
+                    final String key = entry.getKey();
+                    final String value = entry.getValue();
+                    assertEquals("hello " + value, new Interpolator().interpolate("hello $" + key));
+                    assertEquals("hello " + value, new Interpolator().interpolate("hello ${" + key + "}"));
+                    assertEquals("hello " + value, new Interpolator().interpolate("hello ${" + key + ":\"\"}"));
+                    assertEquals("hello " + value, new Interpolator().interpolate("hello ${" + key + ":}"));
+                    assertEquals("hello " + value, new Interpolator().interpolate("hello ${" + key + ":foo}"));
+                    return null;
+                });
     }
 
     @Test

